packages/apputils-extension/src/index.ts

```diff
@@ -335,24 +335,36 @@ export const toggleHeader: JupyterFrontEndPlugin<void> = {
 async function updateTabTitle(workspace: string, db: IStateDB, name: string) {
   const data: any = await db.toJSON();
   let current: string = data['layout-restorer:data']?.main?.current;
-  if (current === undefined) {
+  // File name from current path
+  let currentFile: string | undefined = current
+    ?.split(':')[1]
+    ?.split('\\')
+    ?.pop()
+    ?.split('/')
+    ?.pop();
+  if (currentFile === undefined) {
     document.title = `${PageConfig.getOption('appName') || 'JupyterLab'}${
       workspace.startsWith('auto-') ? ` (${workspace})` : ``
     }`;
   } else {
-    // First 15 characters of current document name
-    current = current.split(':')[1].slice(0, 15);
+    // Truncate to first 12 characters of current document name + ... if length > 15
+    currentFile =
+      currentFile.length > 15
+        ? currentFile.slice(0, 12).concat(`...`)
```

Comment:
Can we use unicode ellipsis (…) here (over just three dots ...) for consistency with other places (and slightly narrower glyph in many fonts)? We may want to see how it renders.

```suggestion
        ? currentFile.slice(0, 12).concat(`…`)
```

Reply:
Yes, sure. I'll check how it renders.